substrate/primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -848,6 +922,11 @@ fn impl_runtime_apis_impl_inner(api_impls: &mut [ItemImpl]) -> Result<TokenStrea
 
 		#runtime_metadata
 
+		pub fn runtime_api_versions() -> #c::ApisVec {
+			let api = #c::vec::Vec::from([RUNTIME_API_VERSIONS.into_owned(), #(#modules::RUNTIME_API_VERSIONS.into_owned()),*]).concat();
+			api.into()
+		}
```

Comment:
I don't really like that `RUNTIME_API_VERSIONS` change its definition from: "all runtime api versions" to "runtime api versions only defined in the main macro". And that people should now use `runtime_api_versions`.

I would prefer to deprecate or remove, rather than changing the definition like this.

Why not handling this on the macro side: `impl_ext` can generate a `PARTIAL_API_VERSIONS` constant and the main macro would then concatenate all the versions in the final constant.

You can concatenate in const like this:
```rust
const A: &[(u32, u32)] = &[(0, 1)];
const B: &[(u32, u32)] = &[(0, 1)];
const C: alloc::borrow::Cow<'static, [(u32, u32)]> = Cow::Borrowed({
	const LEN: usize = A.len() + B.len();
	const ARR: [(u32, u32); LEN] = {
		let mut arr = [(0, 0); LEN];
		// implement with while
		// for &(a, b) in A.iter() {
		// 	arr[i] = (a, b);
		// 	i += 1;
		// }
		// for &(a, b) in B.iter() {
		// 	arr[i] = (a, b);
		// 	i += 1;
		// }

		let mut cursor = 0;
		let mut i = 0;
		while i < A.len() {
			arr[i] = A[i];
			i += 1;
		}
		let mut i = 0;
		while i < B.len() {
			arr[cursor] = B[i];
			cursor += 1;
			i += 1;
		}
		arr
	};
	&ARR
});
```